internal/controller/etcdopstask/reconcile_delete_test.go

```diff
@@ -0,0 +1,277 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package etcdopstask
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	druidapicommon "github.com/gardener/etcd-druid/api/common"
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/core/v1alpha1"
+	"github.com/gardener/etcd-druid/internal/client/kubernetes"
+	"github.com/gardener/etcd-druid/internal/controller/etcdopstask/handler"
+	ctrlutils "github.com/gardener/etcd-druid/internal/controller/utils"
+	druiderr "github.com/gardener/etcd-druid/internal/errors"
+	"github.com/gardener/etcd-druid/test/utils"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/onsi/gomega"
+)
+
+// TestCleanupTaskResources tests the function testCleanupTaskResources
+func TestCleanupTaskResources(t *testing.T) {
+	g := NewGomegaWithT(t)
+	tests := []struct {
+		name                  string
+		task                  *druidv1alpha1.EtcdOpsTask
+		expectedResult        ctrlutils.ReconcileStepResult
+		cleanupFailed         bool
+		resultRequeue         bool
+		expectedLastOperation *druidv1alpha1.LastOperation
+		expectedLastErrors    *druidv1alpha1.LastError
+		expectedPhase         *druidv1alpha1.OperationPhase
+	}{
+		{
+			name:          "Task not found",
+			task:          nil,
+			cleanupFailed: false,
+		},
+		{
+			name:           "Task is in rejected state, last operation is updated",
+			task:           newTestTask(ptr.To(druidv1alpha1.TaskStateRejected)),
+			cleanupFailed:  false,
+			expectedResult: ctrlutils.ContinueReconcile(),
+			expectedLastOperation: &druidv1alpha1.LastOperation{
+				Type:        druidv1alpha1.LastOperationTypeDelete,
+				State:       druidv1alpha1.LastOperationStateProcessing,
+				Description: "Cleanup skipped for rejected task",
+			},
+			expectedPhase: ptr.To(druidv1alpha1.OperationPhaseCleanup),
+		},
+		{
+			name:           "result.requeue is false, no error, last operation is updated",
+			task:           newTestTask(nil),
+			cleanupFailed:  false,
+			expectedResult: ctrlutils.ContinueReconcile(),
+			expectedLastOperation: &druidv1alpha1.LastOperation{
+				Type:  druidv1alpha1.LastOperationTypeDelete,
+				State: druidv1alpha1.LastOperationStateProcessing,
+			},
+			expectedPhase: ptr.To(druidv1alpha1.OperationPhaseCleanup),
+		},
+		{
+			name:           "result.requeue is false, error, last operation and last Error is updated",
```

Comment:
I am not happy with these kinds of unit tests. Why is this mentioned: `last operation and last Error is updated` and updated to what? These tests are not clear and therefore cognitive overload for maintenance. Do not use table driven tests for these and have dedicate unit tests with clear names and description for each test.

Maybe someone else can also review the tests as i think that these tests needs to be rewritten.